src/libs/migrateOnyx.js

```diff
@@ -0,0 +1,31 @@
+import RenameActiveClientsKey from './migrations/RenameActiveClientsKey';
+
+export default function () {
+    const startTime = Date.now();
+    console.debug('[Migrate Onyx] start');
+
+    return new Promise((resolve) => {
+        // Add all migrations to an array so they are executed in order
+        const migrationPromises = [
+            RenameActiveClientsKey,
+        ];
+
+        // Reduce all promises down to a single promise. All promises run in a linear fashion, waiting for the
+        // previous promise to finish before moving onto the next one.
+        migrationPromises
+            /* eslint-disable arrow-body-style */
+            .reduce((previousPromise, migrationPromise) => {
+                return previousPromise.then(() => {
+                    return migrationPromise();
+                });
+            }, Promise.resolve())
```

Comment:
Oh whoa, this is a pretty creative use of `reduce()` that took me a second to wrap my head around. I like it.

Reply:
Haha, yeah, I found it after some digging on StackOverflow. It works well, but sure takes a bit to understand why it works.

Reply:
Very clever, but doesn't [Promise.all](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/all) do this already?

Reply:
`Promise.all` takes an array of promises and will resolve when each one is resolved. But there is no telling when the promises will resolve. They are not dependent on each other but the _responses_ will return in the order that they are passed. If you need to guarantee one after the other then you need this voodoo.

Interestingly, this is probably a perfect case for using async/await since we could just await each migration like this

```js
async function migrate() {
    await migrationOne();
    await migrationTwo();
    await migrationThree();
}
```

Reply:
Yeah, I looked into `Promise.all()` and also `Promise.allSettled()` (which both have slightly different behavior), but ultimately, neither of them will execute the promises in a specific order.

Marc, as much as it pains me to say it... that indeed looks elegant 🙈

Reply:
Oh, we need them to be in order since one migration might depend on the previous one? That makes sense.
Also agree that async/await code looks much better than the current code, should we use it?